filtering.py

```diff
@@ -0,0 +1,941 @@
+import torch
+from torch.utils.data import DataLoader
+import torch.nn as nn
+import itertools
+import warnings
+import model
+import utils
+
+
+# Define basic complex operations on torch.Tensor objects whose last dimension
+# consists in the concatenation of the real and imaginary parts.
+
+def _norm(x):
+    r"""Computes the norm value of a torch Tensor, assuming that it
+    comes as real and imaginary part in its last dimension.
+
+    Parameters:
+    ----------
+    x: torch Tensor [shape=(..., 2)]
+
+    Returns:
+    torch Tensor, with shape as x excluding the last dimension.
+    """
+    return torch.abs(x[..., 0])**2 + torch.abs(x[..., 1])**2
```

Comment:
can be replaced by [`torchaudio.complex_norm`](https://github.com/pytorch/audio/blob/master/torchaudio/functional.py#L469)

Reply:
did you check that?

Reply:
it didn't look so useful to do it imo

Reply:
it didn't look so useful to do it imo

Reply:
maybe their use of torch.pow makes it faster? don't know.

Reply:
its not the same. so this is closed